tests/tests.py

```diff
@@ -542,6 +542,13 @@ def test_ec2(self):
                 'dev').config, 'my-stack-name')
         compare(known, config.ec2())
 
+    def test_ec2_with_no_block_device_specified(self):
+        from testfixtures import compare
+        project_config = ProjectConfig('tests/sample-project.yaml', 'dev')
+        project_config.config.pop('rds')
+        project_config.config['ec2'].pop('block_devices')
+        config = ConfigParser(project_config.config, 'my-stack-name')
+        config.process()
```

Comment:
In this case the behavior is entirely in the `ec2()` function so could you change the test to 1) call that, and 2) check the return value (or at least the devices section) is defaulted appropriately.